viewing_party/party.py

```diff
@@ -1,23 +1,166 @@
-# ------------- WAVE 1 --------------------
+from pprint import pprint
+#HELPER FUNCTION
+def _get_movies_watched_by_friends(user_data):
+    movies_watched_by_friends = []
+    
+    for friend in user_data["friends"]: #[{}]
+        for friend_movie in friend["watched"]: #{}
+            if friend_movie in movies_watched_by_friends:
+                continue
+            else:
+                movies_watched_by_friends.append(friend_movie)
+
+    return movies_watched_by_friends
+
+def _get_movies_not_watched_user(user_data):
+    movies_not_watched_user = []
+    movies_watched_by_friends = _get_movies_watched_by_friends(user_data) 
+
+    for movie in user_data["watched"]:
+        if movie not in movies_watched_by_friends:
+            movies_not_watched_user.append(movie)
+    return  movies_not_watched_user
+
 
+# ------------- WAVE 1 --------------------
 def create_movie(title, genre, rating):
-    pass
+    if not title or not genre or not rating:
+        return None 
+
+    return {"title":title, "genre":genre, "rating":rating}
+
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
 
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        watchlist_title = movie["title"]
+        if title == watchlist_title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            return user_data
+
+    return user_data
+# Note: For Waves 2, 3, 4, and 5, your implementation of each of the functions should not modify user_data.
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    rating = 0
+    num_of_movies = len(user_data["watched"])
+    
+    for movie in user_data["watched"]:
+        rating += movie["rating"]
+
+    if rating == 0:
+        return 0
+    else:
+        average_rating = rating/num_of_movies
+    return average_rating
+        
+        
+
+
+def get_most_watched_genre(user_data):
+    favorite_genre = ""
+    genre_list = []
+    genre = ""
+
+    # If the value of "watched" is an empty list, get_most_watched_genre should return None.
+    if user_data["watched"] == []:
+        return None
 
+    for movie in user_data["watched"]: #this giving each movie dictionary inside the watched list 
+        genre = movie['genre'] #accessing the movie genre and assigning to the genre 
+        genre_list.append(genre) #each genre will be added to the genre_list # the congregate of all genres
 
+
+    favorite_genre = max(set(genre_list), key= genre_list.count) 

```

Comment:
Nice use of max here. I find it useful to also practice writing your own implementation of finding a max value from a dictionary. You may find in an interview if you use built in functions like max that an interviewer may ask you how the function works under the hood. So knowing how to write it on your own can help you better understand the function.